ApiTests/Tests/UI/DynamicControlsTest.cs

```diff
@@ -8,16 +9,30 @@ namespace TestProject.Tests.UI
     internal class DynamicControlsTests : BaseTest
     {
         private static readonly By enableBtn = By.XPath(string.Format(XpathPatterns.preciseTextXpath, "Enable"));
+        private static readonly By input = By.XPath(XpathPatterns.preciseInputXpath);
+        private static readonly string randomValue = Guid.NewGuid().ToString();
 
         [Test]
         public void DynamicControlsTest()
         {
             MainPage mainPage = new MainPage();
             mainPage.ClickOnDynamicControl();
-            Browser.GetDriver().FindElement(enableBtn).Click();
+            var driver = Browser.GetDriver();
+            driver.FindElement(enableBtn).Click();
+
+            var inputElement = Browser.GetDriver().FindElement(input);
+            WebDriverWait wait = new WebDriverWait(driver, TimeSpan.FromSeconds(5));
+            wait.Until(d => inputElement.Enabled);
+
             //assert input is enabled
+            Assert.That(inputElement.Enabled, Is.True);
```

Comment:
`Assert.That(inputElement.Enabled, Is.True, "Input element is not enabled");` would be a better way to use assertion
This message will be displayed in case if the assertion fails and will help to easily locate the issue